pkg/buildsettings/buildSettings.go

```diff
@@ -0,0 +1,101 @@
+package buildsettings
+
+import (
+	"encoding/json"
+	"reflect"
+
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/pkg/errors"
+)
+
+type BuildSettingsInfo struct {
+	Profiles                    []string `json:"profiles,omitempty"`
+	Publish                     bool     `json:"publish,omitempty"`
+	CreateBOM                   bool     `json:"createBOM,omitempty"`
+	LogSuccessfulMavenTransfers bool     `json:"logSuccessfulMavenTransfers,omitempty"`
+	GlobalSettingsFile          string   `json:"globalSettingsFile,omitempty"`
+	DefaultNpmRegistry          string   `json:"defaultNpmRegistry,omitempty"`
+}
+
+type BuildSettings struct {
+	MavenBuild  []BuildSettingsInfo `json:"mavenBuild,omitempty"`
+	NpmBuild    []BuildSettingsInfo `json:"npmBuild,omitempty"`
```

Comment:
the json tag can hold `executeNpmScripts `instead of `npmBuild `and `kanikoExecute `instead of `dockerBuild `so that the final json holds the same name as the step name